src/Aspire.Hosting.Azure/ExistingAzureResourceAnnotation.cs

```diff
@@ -9,16 +9,21 @@ namespace Aspire.Hosting.Azure;
 /// Represents a resource that is not managed by Aspire's provisioning or
 /// container management layer.
 /// </summary>
-public sealed class ExistingAzureResourceAnnotation(ParameterResource nameParameter,
-    ParameterResource? resourceGroupParameter = null) : IResourceAnnotation
+public sealed class ExistingAzureResourceAnnotation(object name, object? resourceGroup = null) : IResourceAnnotation
 {
     /// <summary>
     /// Gets the name of the existing resource.
     /// </summary>
-    public ParameterResource NameParameter { get; } = nameParameter;
+    /// <remarks>
+    /// Supports a <see cref="string"/> or a <see cref="ParameterResource"/> via runtime validation.
+    /// </remarks>
+    public object Name { get; } = name;
```

Comment:
Can we instead just have the new overload call the `AddParameter` method that takes a hard-coded string value?

Then everything is built on Parameters underneath, but the API can take `string` or `Parameter`.

Reply:
https://github.com/dotnet/aspire/blob/3c823d994e1f4d16fef1ede4b7e3fc430ea7e00a/src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs#L50-L57

is the overload I'm talking about.

Reply:
We'd need to figure out if the `Name` pointed to an _actual_ `ParameterResource` that needed to be mapped to a provisioning parameter in the Bicep/manifest or a faux one where we could just extract the value. Is there a way to do that with `AddParameter`?

Reply:
Would it be terrible if it always created the parameter underneath (in the manifest and bicep) and the parameter's value was hard-coded in the manifest?

Reply:
`azd` has some special behavior that will provide an enhanced prompt for selecting the resource group when we detect it as a parameter in the resource group. I suspect that mapping it this way will cause that logic to trigger when it shouldn't?

Reply:
I don't think that's the end of the world, but it's no different from what we do with environment variables, arguments, bicep parameters etc.